lib/gui/app/models/storage.js

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2018 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const INDENTATION_SPACES = 2
+
+/**
+ * @summary Localstorage class and helper functions
+ * @class
+ * @public
+ */
+class Storage {
+  /**
+   * @function
+   * @public
+   *
+   * @param {String} superkey - superkey
+   *
+   * @example
+   * const potatoStorage = new Storage('potato')
+   */
+  constructor (superkey) {
+    this.superkey = superkey
+  }
+
+  /**
+   * @summary Get the whole object under the superkey
+   * @function
+   * @public
+   *
+   * @returns {Object}
+   *
+   * @example
+   * for (const key in potatoStorage.getAll()) {
+   *   console.log(key)
+   * }
+   */
+  getAll () {
+    try {
+      // JSON.parse(null) === null, so we fallback to {}
+      return JSON.parse(window.localStorage.getItem(this.superkey)) || {}
+    } catch (err) {
+      return {}
```

Comment:
Not sure if this would be a good way to handle errors. In which situation would we expect this to fail? And if it fails, this would effectively clear all settings to default values once .setAll() is called – which in turns seems rather unsafe.

Reply:
Yeah, what @lurch said above is the error case I had in mind. I assume the case where localStorage fails in another way is extremely rare (but probably not impossible?), but how do you suggest we handle any potential errors more gracefully here? I guess we could check if it's the most common mistake, a parse error, and then return an empty object if so, and in any other error case we unleash the error upon whatever code uses it?

Reply:
@Shou @lurch nothing apart from the code we run would have access to localStorage (unless the user willingly does something in the devtools), so imo this should explode if it fails, as .setAll() does

Reply:
How about this commit? We reset the value to an empty object and propagate the error subsequently so we don't allow the user to get stuck with a broken Etcher